src/sentry/api/endpoints/project_metrics_extraction_rules.py

```diff
@@ -104,22 +108,27 @@ def post(self, request: Request, project: Project) -> Response:
                     configs.append(
                         SpanAttributeExtractionRuleConfig.from_dict(obj, request.user.id, project)
                     )
+
+                validate_number_of_extracted_metrics(project)
+
                 schedule_invalidate_project_config(
                     project_id=project.id, trigger="span_attribute_extraction_configs"
                 )
 
-        except Exception:
-            sentry_sdk.capture_exception()
-            return Response(
-                status=400,
+            persisted_config = serialize(
+                configs,
+                request.user,
+                SpanAttributeExtractionRuleConfigSerializer(),
             )
+            return Response(data=persisted_config, status=200)
 
-        persisted_config = serialize(
-            configs,
-            request.user,
-            SpanAttributeExtractionRuleConfigSerializer(),
-        )
-        return Response(data=persisted_config, status=200)
+        except MetricsExtractionRuleValidationError as e:
+            logger.warning("Failed to update extraction rule", exc_info=True)
```

Comment:
Is there a reason why are we logging this?
If we want to count how many times it is happening, is there a reason why aren't we using Sentry for this?